pkg/query-service/rules/promRule.go

```diff
@@ -41,42 +41,49 @@ type PromRule struct {
 	active map[uint64]*Alert
 
 	logger log.Logger
+	isTest bool
```

Comment:
I suggest this shouldn't know if the rule is for testing purpose or not. May be a generic like `forceSend` or `sendAlways` for short circuiting the original logic and testing APIs can use it. What do you think?

Reply:
I thought about initially. but  I opted out since test not just does a force send but also alters the message as well.  further, the force send only occurs in thresholdRule and not in PromQL. 

The use of isTest kind of makes it easy for the maintainer to know specific logic tied to TestingAPI as well.  It also accomodates future modifications to this condition (e.g. changes to the message, mocking resolution alert etc) 

Let me know if you feel strongly about this. I can consider making changes.